build/commands/scripts/lit_mangler.ts

```diff
@@ -0,0 +1,200 @@
+    for (const child of template.children) {
+        const original = child.text
+        child.text = injectPlaceholders(child)
+        template.text = template.text.replace(original, `$$$$lit_mangler_${child.id}$$$$`)
+    }
+
+    return template.text
+}
+
+const replacePlaceholders = (template: HTMLTemplateTags) => {
+    for (const child of template.children) {
+        child.text = replacePlaceholders(child)
+        template.text = template.text.replaceAll(`$$lit_mangler_${child.id}$$`, child.text)
+    }
+
+    return template.text
+}
+
+let result: HTMLTemplateTags | undefined = undefined
+
+const load = (file: string) => {
+    if (result !== undefined) {
+        throw new Error('This should only be called once per file')
+    }
+
+    result = getLiteralsFromFile(file)
+    injectPlaceholders(result)
+}
+
+const write = (file: string) => {
+    if (!result) {
+        throw new Error('This should only be called after load')
+    }
+
+    const text = replacePlaceholders(result)
+    writeFileSync(file, text)
+}
+
+/**
+ * Mangles a given html template using the given mangler function.
+ *
+ * Example usage:
+ * mangle((element) => element.textContent = "foo", t => t.text.includes('allow-incognito'))
+ *
+ * @param mangler The function to use to mangle the template.
+ * @param getTemplate The template to mangle, or a predicate to find a matching template. If undefined, the first html template will be used.
+ */
+export const mangle = (mangler: (element: DocumentFragment) => void, getTemplate?: HTMLTemplateTags | ((template: HTMLTemplateTags) => boolean)) => {
+    if (!result) {
+        throw new Error('This should only be called after load!')
+    }
+
+    let template: HTMLTemplateTags | undefined = undefined
+    if (!getTemplate) {
+        template = findTemplate(() => true)
+    } else if (typeof getTemplate === 'function') {
+        template = findTemplate(getTemplate)
+    } else {
+        template = getTemplate
+    }
+
+    // If we don't have a template throw an error - mangling isn't going to work.
+    if (!template) {
+        throw new Error(`Failed to find template matching ${getTemplate}. If this was working before upstream may have changed.`)
+    }
+
+    const world = new JSDOM()
+    const document = world.window.document
+    const element = document.createElement('template')
+    element.innerHTML = template.text
+        // Note: we ensure that all attributes are quoted so the template can be
+        // parsed as HTML.
+        // This won't work for attributes with braces in them, but it'll
+        // drastically break the HTML output from upstrean, so it should be
+        // pretty obvious that something went wrong.
+        .replaceAll(/(\w+)(\s+)?=(\s+)?(\$\{.*?\})(\s|>)/gi, (...args) => {
+            return `${args[1]}="${args[4].replaceAll('"', '&quot;')}"${args[5]}`
+        })
+
+    mangler(element.content)
+
+    template.text = element.innerHTML
```

Comment:
this does not look right

Reply:
I think it is because there's an encode which happens when parsing to HTML and we want things to roundtrip nicely:

```
// the string
"<div>${"<foo&"}</div>"

// is parsed to
<div>${&quot;&lt;foo&amp;&quot}</div>

// which should roundtrip back to the string
"<div>${"<foo&"}</div>"

```

This is because its valid to have `html` which is really just a string inside the interpolated content (in the real world it gets replaced with the result of parsing to JS). However, we want the raw string from inside the interpolation so we can edit it.